core/domain/recommendations_services.py

```diff
@@ -147,7 +147,9 @@ def save_topic_similarities(
     topic_similarities_entity.update_timestamps()
     topic_similarities_entity.put()
 
-    # Ruling out the possibility of None for mypy type checking.
+    # Ruling out the possibility of None for mypy type checking, because
```

Comment:
A better way to do this is probably:

    retrieved_topic_similarities_entity = (
        recommendations_models.TopicSimilaritiesModel.get(
            recommendations_models.TOPIC_SIMILARITIES_ID, strict=False))
    topic_similarities_entity = (
        retrieved_topic_similarities_entity 
        if retrieved_topic_similarities_entity is not None
        else ...)

and then you shouldn't need this assert.

Reply:
Done